text/0000-formatted-bliss.md

```diff
@@ -0,0 +1,226 @@
+>
+> Furthermore, by sticking to an established style guide (such as this one), programmers don't need to formulate ad hoc style rules, nor do they need to debate with other programmers what style rules should be used, saving time, communication overhead, and mental energy.
+>
+> Humans comprehend information through pattern matching. By ensuring that all Rust code has similar formatting, less mental effort is required to comprehend a new project, lowering the barrier to entry for new developers.
+>
+> Thus, there are productivity benefits to using a formatting tool (such as rustfmt), and even larger benefits by using a community-consistent formatting, typically by using a formatting tool's default settings.
+
+Simply put, we wish to unlock these benefits for the repository.
+In particular, as the repository is large and is read and written
+to daily by many developers, we expect the benefits in terms of a
+consistent style to be large.
+
+Moreover, to lower the barrier to entry for working on compiler development,
+we wish to adopt the standard Rust Style Guide as much as is possible.
+This will ensure that most Rust developers feel at home with style inside the repository.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+See the [summary] for a brief explanation or read below for a detailed technical
+description of what this proposal entails.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+To disrupt normal daily development as little as possible in the repository,
+the proposal has been divided into steps which in part can and should be
+executed independently of each other.
+
+## 1. Introducing *the style*
+
+As noted in the [motivation], we wish to make the repository where any Rust
+developer feels at home. To that end, we introduce and adopt the standard
+Rust Style Guide into the repository.
+
+*The style* is defined in terms of a `rustfmt.toml` file (hence: *the style file*):
+
+```toml
+version = "Two"
+use_small_heuristics = "Max"
+```
+
+The style file is located at the top of the repository.
+
+This step has already been completed.
+
+## 2. Enforce the style
+
+The `tidy` tool, defined in `src/tools/tidy`, is run each time a new PR is created
+or when a commit is added to said PR. The tool is also run when a PR,
+approved for merging with e.g. `@bors r=reviewer`, is tested for merging.
+
+To `tidy`, new logic is added that checks whether the code is conformant with
+the style. This will result in a PR being rejected when changes in said PR does
+not conform to the style.
+
+To facilitate incremental enforcement of the style and to disrupt development
+minimally inside the repository, the `tidy` tool will only enforce the style
+in folders with an `.enforcefmt` file. In such folders, the style will be
```

Comment:
I would personally prefer a list in tidy itself rather than dozens of these files strewn around the repository, but I don't care too much.

Reply:
Implementors prerogative? I was thinking `.enforcefmt` would be easier for the transition since you can incrementally just "move up" the `.enforcefmt` file instead of having to fiddle with tidy.